bundle/config/mutator/resolve_variable_references.go

```diff
@@ -87,6 +95,33 @@ func (m *resolveVariableReferences) Apply(ctx context.Context, b *bundle.Bundle)
 
 	var diags diag.Diagnostics
 
+	for round := range 1 + m.extraRounds {
+		hasUpdates, newDiags := m.resolveOnce(b, prefixes, varPath)
+
+		diags = diags.Extend(newDiags)
+
+		if diags.HasError() {
+			break
+		}
+
+		if !hasUpdates {
+			break
+		}
+
+		if round >= maxResolutionRounds-1 {
+			diags = diags.Append(diag.Diagnostic{
```

Comment:
shall we break here as well?

Reply:
Well it's the last stmt in the last loop iteration, so it does not matter.

Reply:
it's not necessarily last one if I pass extraRounds bigger than maxResolutionRounds

Reply:
good catch! it's a bug.  c3ac311e9c48f2059a05a4794fe84529e64acc8e